tocgen.py

```diff
@@ -0,0 +1,20 @@
+from jinja2 import Environment, FileSystemLoader
+import os, re
+
+env = Environment(loader=FileSystemLoader(os.path.dirname('./')))
+template = env.get_template('toc.md.j2')
+
+rootDir = '.'
+
+list_dirs = ""
+for dirName, subdirList, fileList in os.walk(rootDir):
+    if re.match('^./.git',dirName):
+        pass
+    else:
+        for fname in fileList:
+            if fname == "README.md":
+                list_dirs += "* [{}](#{})\n".format(dirName,os.path.join(dirName,fname).replace(" ", "%"))
```

Comment:
This line is doing too much (6+ different operations) - break it out into multiple steps.